taipy/core/_manager/_manager_factory.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2022 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from importlib import util
+from typing import Type
+
+from taipy.core.common._utils import _load_fct
+from taipy.core.cycle._cycle_manager import _CycleManager
+from taipy.core.data._data_manager import _DataManager
+from taipy.core.job._job_manager import _JobManager
+from taipy.core.pipeline._pipeline_manager import _PipelineManager
+from taipy.core.scenario._scenario_manager import _ScenarioManager
+from taipy.core.task._task_manager import _TaskManager
+
+
+class _ManagerFactory:
+    _TAIPY_ENTERPRISE_MODULE = "taipy.enterprise"
+    _TAIPY_ENTERPRISE_CORE_MODULE = _TAIPY_ENTERPRISE_MODULE + ".core"
+
+    @classmethod
+    def _scenario_manager(cls) -> Type[_ScenarioManager]:
+        if cls._has_enterprise():
+            return _load_fct(
+                cls._TAIPY_ENTERPRISE_CORE_MODULE + ".scenario._authorized_scenario_manager",
```

Comment:
I would name the module in the enterprise version _scenarion_manager and not _authorized_scenario_manager. The purpose is to hide from the community version what is done in the enterprise version, and how it is implemented.
```suggestion
                cls._TAIPY_ENTERPRISE_CORE_MODULE + ".scenario._scenario_manager",
```

Reply:
Isn't it counterintuitive to not let the community know what we provide in the enterprise version?

Reply:
i think the idea is to keep people from copying our product :D

Reply:
exactly